x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/mistral/embeddings/MistralEmbeddingsServiceSettings.java

```diff
@@ -0,0 +1,208 @@
+import org.elasticsearch.xpack.inference.services.ServiceUtils;
+import org.elasticsearch.xpack.inference.services.settings.FilteredXContentObject;
+import org.elasticsearch.xpack.inference.services.settings.RateLimitSettings;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.TransportVersions.ADD_MISTRAL_EMBEDDINGS_INFERENCE;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.DIMENSIONS;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.MAX_INPUT_TOKENS;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.SIMILARITY;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalBoolean;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalPositiveInteger;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractRequiredString;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractSimilarity;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.removeAsType;
+import static org.elasticsearch.xpack.inference.services.mistral.MistralConstants.DIMENSIONS_SET_BY_USER;
+import static org.elasticsearch.xpack.inference.services.mistral.MistralConstants.MISTRAL_MODEL_FIELD;
+
+public class MistralEmbeddingsServiceSettings extends FilteredXContentObject implements ServiceSettings {
+    public static final String NAME = "mistral_embeddings_service_settings";
+
+    private final String model;
+    private final Integer dimensions;
+    private final Boolean dimensionsSetByUser;
+    private final SimilarityMeasure similarity;
+    private final Integer maxInputTokens;
+    private final RateLimitSettings rateLimitSettings;
+
+    // default for Mistral is 5 requests / sec
+    // setting this to 240 (4 requests / sec) is a sane default for us
+    protected static final RateLimitSettings DEFAULT_RATE_LIMIT_SETTINGS = new RateLimitSettings(240);
+
+    public static MistralEmbeddingsServiceSettings fromMap(Map<String, Object> map, ConfigurationParseContext context) {
+        ValidationException validationException = new ValidationException();
+
+        String model = extractRequiredString(map, MISTRAL_MODEL_FIELD, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        SimilarityMeasure similarity = extractSimilarity(map, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        Integer maxInputTokens = extractOptionalPositiveInteger(
+            map,
+            MAX_INPUT_TOKENS,
+            ModelConfigurations.SERVICE_SETTINGS,
+            validationException
+        );
+        RateLimitSettings rateLimitSettings = RateLimitSettings.of(map, DEFAULT_RATE_LIMIT_SETTINGS, validationException);
+
+        Integer dims = removeAsType(map, DIMENSIONS, Integer.class);
+        Boolean dimensionsSetByUser = extractOptionalBoolean(map, DIMENSIONS_SET_BY_USER, validationException);
```

Comment:
It doesn't look like we use `dimensionsSetByUser` in the actual request to mistral. Do we need it? I think we'd only need it if we need to send the `dimensions` field to mistral. I also don't see it in their docs: https://docs.mistral.ai/api/#operation/createEmbedding

Reply:
If it's not needed, that's great... I thought it was needed however for our processing on the results? (I'm probably wrong...)

Reply:
Yeah I think we only need it if we have to send the `dimensions` field to mistral. We use it in the service but that's only to verify that we got the right number of dimensions back when we send it to the external provider.